[Low Jun Yu] iP

jylow opened the pull request:
### B055man :smile:

<blockquote>
"I will always find a lazy person to do the hardest tasks" - Bill Gates
</blockquote>

B055man is a friendly chatbot to help keep track of things that you need to complete. It is,

* User Friendly
* Simple to pickup
* ~$29.99~ ****FREE**** to use :exclamation:

All you need to use it,
1. Have a computer
2. Click [here](https://github.com/jylow/ip/releases/tag/A-Jar) to download (I promise its not malware) :rofl:
3. Run the program
4. Use B055man to manage your tasks

Did I mention that its ****free****?

!Features of B055man:!!
- [ ] Manage tasks
- [X] Add todos, events and deadlines
- [ ] Saves previously entered tasks 

Make sure that you have downloaded the right code:
Below is the `main` method that you should see at the bottom of the `Duke` class
```ruby
public static void main(String[] args) {
        new Duke("data/tasks.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

jylow committed:
Level-0

jylow committed:
Level-0

jylow committed:
Level-0

jylow committed:
Level-1

jylow committed:
Level-2

jylow committed:
Level-3

jylow committed:
Level-4

jylow committed:
A-TextUiTesting

jylow committed:
Level-5

jylow committed:
Level-5

jylow committed:
Level-6

jylow committed:
Level-7

jylow committed:
Level-7

jylow committed:
Merge pull request #1 from jylow/branch-Level-7

Level-7

jylow committed:
Level-8

jylow committed:
Merge pull request #2 from jylow/branch-Level-8

Level-8

jylow committed:
A-MoreOOP

jylow committed:
A-MoreOOP

jylow committed:
Merge pull request #4 from jylow/branch-A-MoreOOP

Add better OOP design

jylow committed:
A-Packages

jylow committed:
Merge pull request #5 from jylow/branch-A-Packages

Sort classes into packages

jylow committed:
A-Gradle

jylow committed:
Merge pull request #6 from jylow/add-gradle-support

Add gradle support

jylow committed:
A-JUnit

jylow committed:
Merge pull request #7 from jylow/branch-A-JUnit

Add JUnit tests

jylow committed:
A-Jar

jylow committed:
Merge pull request #8 from jylow/branch-A-Jar

Add A-Jar

jylow committed:
A-JavaDoc

jylow committed:
Level-9

jylow committed:
Merge pull request #10 from jylow/branch-A-JavaDoc

Add JavaDoc

jylow committed:
Merge pull request #9 from jylow/branch-Level-9

Add find feature

jylow committed:
A-CodingStandard

jylow committed:
Merge pull request #11 from jylow/branch-A-CodingStandard

Add coding standard

jylow committed:
A-Checkstyle

jylow committed:
A-Level10

jylow committed:
Revert "A-Checkstyle"

This reverts commit 7e5a55b5e7c7cdf35a22ed5a328574afbfebf299.

howenc commented on src/main/java/duke/util/TaskList.java:
```diff
@@ -0,0 +1,166 @@
+        }
+        tasklst.get(taskNumber - 1).markAsDone();
+        ui.showMarked(tasklst.get(taskNumber - 1));
+    }
```
I don't like how ui is only included in the arguments just to showMarked(Task task).
Do you think making the Ui methods static would help make this code simpler?

howenc commented on src/main/java/duke/util/TaskList.java:
```diff
@@ -0,0 +1,166 @@
+            tasklst.get(taskNumber - 1).unmarkAsDone();
+            ui.showUnmarked(tasklst.get(taskNumber - 1));
+        }
+    }
```
Same here, I don't like that ui is here.
Should be a better way to do this right?

jylow committed:
A-CheckStyle

jylow committed:
Merge pull request #12 from jylow/branch-A-CheckStyle

Add check style

howenc commented on src/main/java/duke/util/TaskList.java:
I like the logic in this file.
It is very smooth and flows nicely.
Each method is short and concise except for that 'parser' you have within the constructor.
What I can think of is putting it in another class just to parse the data you load from the file, but that may be too much work.
The next is that I strongly believe that if you change Ui to have static methods, you would not have to deal with needing to place it in the arguments.

howenc commented on src/main/java/duke/task/Task.java:
Short and sweet I like it.

howenc commented on src/main/java/duke/task/Deadline.java:
Just wondering, you are adding the format with YYYY-MM-DD HHHH, but printing the format in dd MMM yyyy?
I think it would be better to be more consistent?
But if it is a design choice then more power to you.

howenc commented on src/main/java/duke/task/Event.java:
I see `DateTimeFormatter.ofPattern("dd MMM yyyy")` a lot, are you able to make it inheritable or referred to easily?

howenc commented on src/main/java/duke/command/ExitCommand.java:
```diff
@@ -0,0 +1,36 @@
+     * Checks if the current task is an exit task.
+     * @return true since task is an exit task
+     */
+    @Override
+    public boolean isExit() {
+        return true;
+    }
+}
```
I don't like this.
If you just need to return a boolean, do you think letting execute return a boolean is good enough?

howenc reviewed (commented):
Overall it is good, I just did some nit-picking.
The logic is sound and reasonable.

jylow committed:
A-Level10

jylow committed:
Merge pull request #13 from jylow/branch-Level-10

Branch level 10

jylow committed:
Assert

jylow committed:
Clean up code

jylow committed:
Merge pull request #15 from jylow/branch-A-Assertions

Add insertions

jylow committed:
Merge pull request #17 from jylow/master

parallel branch practice

jylow committed:
Merge pull request #16 from jylow/branch-A-CodeQuality

Clean up code

jylow committed:
Improve list filter

jylow committed:
Merge pull request #18 from jylow/branch-C-BetterSearch

Improve list filter

jylow committed:
Add priority feature

jylow committed:
Implement priority tracking of tasks in chatbot.

This is to enable users to track the priority of
completing tasks.

Priority enum is used to ensure that the priority
levels of the task falls within specific range.
SetPriority command is also added to handle the
update of priority.

Another implementation is the ability to save the
priority level in the text file so that its status
can be retrieved in the future.

Further extension can be done to give functionality
to the different priority levels.

jylow committed:
Revert "Add priority feature"

This reverts commit 4cee3c03c063901a8c9f2d94f30c4f114ad0d1d4.

jylow committed:
Implement priority tracking of tasks in chatbot.

This is to enable users to track the priority of
completing tasks.

Priority enum is used to ensure that the priority
levels of the task falls within specific range.
SetPriority command is also added to handle the
update of priority.

Another implementation is the ability to save the
priority level in the text file so that its status
can be retrieved in the future.

Further extension can be done to give functionality
to the different priority levels.

jylow committed:
Merge pull request #20 from jylow/branch-C-Priority

Branch c priority

jylow committed:
Change the java doc comments of the code.

Changes to the comments are done as flagged out
by the auto grader.

Helps make the code more readable.

jylow committed:
Merge pull request #21 from jylow/branch-updateCode

Change the java doc comments of the code.

jylow committed:
Add user guide to the docs folder as well as Ui screenshot
to provide a base for a website to be displayed to the user
as guidance on how to utilise the chatbot.

Changes to the Ui messages to improve formatting of the
responses. This is to help make more consistent
formatting of display to improve user experience

jylow committed:
Add user guide to the docs folder as well as Ui screenshot
to provide a base for a website to be displayed to the user
as guidance on how to utilise the chatbot.

Changes to the Ui messages to improve formatting of the
responses. This is to help make more consistent
formatting of display to improve user experience

jylow committed:
Merge pull request #22 from jylow/branch-A-UserGuide

Add user guide

jylow committed:
Changing of filepath done to ensure that filepath
is relative. Changes to code to ensure that new folder
is created when there is none to store the textfile

Change was done so that when using jar file, the
new textfile will be able to be written into and
accessed by the user in the instance of a lack of
a data folder.

Another change is to the User guide to make it easier
to read

jylow committed:
Merge pull request #23 from jylow/branch-filePathFix

Changing of filepath done to ensure that filepath

jylow committed:
change to build.gradle file to
enable cross-platform capabilities

jylow committed:
Merge pull request #24 from jylow/branch-A-debugging

change to build.gradle

jylow committed:
Changes to code to improve SLAP of code.

This is to make abstraction neater and
easier for others to read.

Changes made to how new folder and file is
created by shifting methods from Duke to
Storage which should handle all file
interactions

jylow committed:
Merge pull request #25 from jylow/branch-improveSlap

Changes to code to improve SLAP of code.

jylow committed:
Changes to code to improve SLAP of code.

This is to make abstraction neater and
easier for others to read.

Changes made to how new folder and file is
created by shifting methods from Duke to
Storage which should handle all file
interactions

jylow committed:
Added Ui image along with profile
images

jylow committed:
Merge pull request #26 from jylow/branch-A-UserGuide

Added Ui image along with profile

jylow committed:
Merge pull request #27 from jylow/branch-improveSlap

Changes to code to improve SLAP of code.
